src/Services/ReadOnlyCacheService.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace SilverStripe\Assets\Services;
+
+use SilverStripe\Core\Injector\Injectable;
+
+/**
+ * Used to cache results for the duration of a request during read-only file operations
+ * Do not use this during any create, update or delete operations
+ */
+class ReadOnlyCacheService
```

Comment:
Maybe this should implement the `Flushable` interface?

Reply:
I don't think we should officially support this API. Mark it as `@internal`.

Reply:
It's not flushable, there's nothing to flush.  The cache is only used for a single request and then it's gone

Reply:
[`Resettable`](https://api.silverstripe.org/5/SilverStripe/Core/Resettable.html) is the appropriate interface for classes which have in-memory caches.